### PR TITLE
Added SpanExporterFactory and Config

### PR DESCRIPTION
**Pull request by prydin**
Added interfaces for creating SpanExporters from auto instrumentation agents according to discussion in  #849.

### CODE REVIEW

File: sdk_contrib/auto_config/src/main/java/io/opentelemetry/sdk/contrib/auto/config/SpanExporterFactory.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.contrib.auto.config;
+
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+
+/**
+ * A {@link SpanExporterFactory} acts as the bootstrap for a {@link SpanExporter} implementation. An
+ * exporter must register its implementation of a {@link SpanExporterFactory} through the Java SPI
```
Comment (thisthat) [inline sdk_contrib/auto_config/src/main/java/io/opentelemetry/sdk/contrib/auto/config/SpanExporterFactory.java]: I am not sure about the SPI framework. In my understanding, the implementation should register the exporter via the API offered by the SDK.

File: sdk_contrib/auto_config/src/main/java/io/opentelemetry/sdk/contrib/auto/config/Config.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2019, OpenTelemetry Authors
```
[the remark this answers is not in the document]
Comment (prydin) [reply sdk_contrib/auto_config/src/main/java/io/opentelemetry/sdk/contrib/auto/config/Config.java]: Ooops! :) Fixing that.

File: sdk_contrib/auto_config/src/main/java/io/opentelemetry/sdk/contrib/auto/config/Config.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ * mapping directly to their native configuration framework.
+ *
+ * <p>This interface is intentionally kept very simple since the underlying implementations may only
+ * have access to very basic configuration mechanisms such as system properties and environment
+ * variables.
+ */
+public interface Config {
+  /**
+   * Returns the string configuration property corresponding to a key. If the underlying
+   * implementation cannot find a property for the key, <code>defaultValue</code> is returned.
```
[the remark this answers is not in the document]
Comment (prydin) [reply sdk_contrib/auto_config/src/main/java/io/opentelemetry/sdk/contrib/auto/config/Config.java]: OK. No problem.

File: sdk_contrib/auto_config/src/main/java/io/opentelemetry/sdk/contrib/auto/config/SpanExporterFactory.java (hunk 1 above)
Comment (prydin) [reply sdk_contrib/auto_config/src/main/java/io/opentelemetry/sdk/contrib/auto/config/SpanExporterFactory.java]: Well, this is for the auto instrumentation. The exporter and the auto instrumenter implementations don't know about each other, so we use SPI to load the exporter dynamically based on a JAR file name that's passed. 

Have a look here: https://github.com/open-telemetry/opentelemetry-auto-instr-java/blob/239eb53a44000f9f77e251a79e7189883451f5fc/agent-tooling/src/main/java/io/opentelemetry/auto/tooling/TracerInstaller.java#L41

Comment (thisthat) [reply sdk_contrib/auto_config/src/main/java/io/opentelemetry/sdk/contrib/auto/config/SpanExporterFactory.java]: Thanks for the clarification :+1:

**Pull request merged**